airflow/models/baseoperator.py

```diff
@@ -767,6 +767,9 @@ def __init__(
         dag = dag or DagContext.get_current_dag()
         task_group = task_group or TaskGroupContext.get_current_task_group(dag)
 
+        if dag is not None and dag.fail_stop and trigger_rule != DEFAULT_TRIGGER_RULE:
+            raise DagInvalidTriggerRule()
```

Comment:
It seems awkward to me `fail_stop` and `DEFAULT_TRIGGER_RULE` are repeated when we check to raise the exception, and when the exception is rendered.

Would something like a `check` function on `DagInvalidTriggerRule` encapsulating the logic be a good idea, so the logic exists only in one class?

Reply:
That makes sense. I have added a static `check` method to the `DagInvalidTriggerRule` class, which checks the necessary condition and throws an exception if needed. Let me know if this was what you had in mind or if there is a better possible implementation for this.